code/modules/wiremod/shell/mech.dm

```diff
@@ -0,0 +1,203 @@
+/obj/vehicle/sealed/mecha/combat/durand/shell
+	desc = "An aging combat exosuit utilized by the Nanotrasen corporation, altered for autonomous control by circuits."
+	name = "\improper Circutix"
+
+/obj/vehicle/sealed/mecha/combat/durand/shell/Initialize()
+	. = ..()
+	AddComponent( \
+		/datum/component/shell, \
+		unremovable_circuit_components = list(new /obj/item/circuit_component/mech_movement, new /obj/item/circuit_component/mech_equipment), \
+		capacity = SHELL_CAPACITY_LARGE, \
+		shell_flags = SHELL_FLAG_ALLOW_FAILURE_ACTION \
+	)
+
+///Only exists to make porting a pain when any tg cooder realizes what a piece of shit this is
+/obj/vehicle/sealed/mecha/combat/durand/shell/proc/attack_target(atom/target, params)
+	SIGNAL_HANDLER
+	var/obj/vehicle/sealed/mecha/self = src
+	if(!isturf(target) && !isturf(target.loc)) // Prevents inventory from being drilled
+		return
+	if(self.completely_disabled || self.is_currently_ejecting || (self.mecha_flags & CANNOT_INTERACT))
+		return
+	//var/list/modifiers = params2list(params)
+	if(!self.get_charge())
+		return
+	if(src == target)
+		return
+	var/dir_to_target = get_dir(src,target)
+	if(dir_to_target && !(dir_to_target & dir))//wrong direction
+		return
+	if(self.internal_damage & MECHA_INT_CONTROL_LOST)
+		target = pick(view(3,target))
+
+
+	if(self.selected)
+		if(!Adjacent(target) && (self.selected.range & MECHA_RANGED))
+			if(SEND_SIGNAL(src, COMSIG_MECHA_EQUIPMENT_CLICK, src, target) & COMPONENT_CANCEL_EQUIPMENT_CLICK)
+				return
+			INVOKE_ASYNC(self.selected, /obj/item/mecha_parts/mecha_equipment.proc/action, src, target, params)
+			return
+		if((self.selected.range & MECHA_MELEE) && Adjacent(target))
+			if(SEND_SIGNAL(src, COMSIG_MECHA_EQUIPMENT_CLICK, src, target) & COMPONENT_CANCEL_EQUIPMENT_CLICK)
+				return
+			INVOKE_ASYNC(self.selected, /obj/item/mecha_parts/mecha_equipment.proc/action, src, target, params)
+			return
+	var/on_cooldown = TIMER_COOLDOWN_CHECK(src, COOLDOWN_MECHA_MELEE_ATTACK)
+	var/adjacent = Adjacent(target)
+
+	var/mob/living/attacker = new /mob/living/
+	attacker.combat_mode = TRUE
```

Comment:
if you really need this, just keep the attacker in a variable on the mech type. Create it in Initialize and remove in Destroy